### ISSUE TITLE
Occasional InvalidTokenError in AppComponent unit test [🐛]

### ISSUE THREAD

**Issue by bkimminich** [juice-shop/juice-shop#1263]
The error below is _occasionally_ thrown on Travis-CI during unit testing:

```
e 78.0.3904 (Mac OS X 10.13.3) AppComponent should create the app FAILED
	InvalidTokenError: Invalid token specified: Cannot read property 'replace' of undefined
	error properties: Object({ ngDebugContext: DebugContext_({ view: Object({ def: Object({ factory: Function, nodeFlags: 668647443, rootNodeFlags: 33554433, nodeMatchedQueries: 4610, flags: 0, nodes: [ Object({ nodeIndex: 0, parent: null, renderParent: null, bindingIndex: 0, outputIndex: 0, checkIndex: 0, flags: 33554433, childFlags: 608223235, directChildFlags: 4243457, childMatchedQueries: 2, matchedQueries: Object({  }), matchedQueryIds: 0, references: Object({  }), ngContentIndex: null, childCount: 6, bindings: [ Object({ flags: 2, ns: '', name: 'mat-toolbar-multiple-rows', nonMinifiedName: 'mat-toolbar-multiple-rows', securityContext: undefined, suffix: undefined }), Object({ flags: 2, ns: '', name: 'mat-toolbar-single-row', nonMinifiedName: 'mat-toolbar-single-row', securityContext: undefined, suffix: undefined }) ], bindingFlags: 2, outputs: [  ], element: Object({ ns: '', name: 'mat-toolbar', attrs: [ Array, Array ], template: null, componentProvider: Object({ nodeIndex: 1, parent: <circular re ...
	    at <Jasmine>
	    at Object.../node_modules/jwt-decode/lib/index.js (http://localhost:9876/_karma_webpack_/vendor.js:112:31)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/app.guard.ts (http://localhost:9876/_karma_webpack_/main.js:3997:15257)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/navbar/navbar.component.ts (http://localhost:9876/_karma_webpack_/main.js:6266:28135)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/app.component.spec.ts (http://localhost:9876/_karma_webpack_/main.js:3904:28)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at webpackContext (http://localhost:9876/_karma_webpack_/main.js:960:9)
	    at <Jasmine>
	Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
	    at <Jasmine>
HeadlessChrome 78.0.3904 (Mac OS X 10.13.3): Executed 476 of 508 (1 FAILED) (skipped 1) (0 secs / 1 min 32.311 secs)
HeadlessChrome 78.0.3904 (Mac OS X 10.13.3) AppComponent should create the app FAILED
	InvalidTokenError: Invalid token specified: Cannot read property 'replace' of undefined
	error properties: Object({ ngDebugContext: DebugContext_({ view: Object({ def: Object({ factory: Function, nodeFlags: 668647443, rootNodeFlags: 33554433, nodeMatchedQueries: 4610, flags: 0, nodes: [ Object({ nodeIndex: 0, parent: null, renderParent: null, bindingIndex: 0, outputIndex: 0, checkIndex: 0, flags: 33554433, childFlags: 608223235, directChildFlags: 4243457, childMatchedQueries: 2, matchedQueries: Object({  }), matchedQueryIds: 0, references: Object({  }), ngContentIndex: null, childCount: 6, bindings: [ Object({ flags: 2, ns: '', name: 'mat-toolbar-multiple-rows', nonMinifiedName: 'mat-toolbar-multiple-rows', securityContext: undefined, suffix: undefined }), Object({ flags: 2, ns: '', name: 'mat-toolbar-single-row', nonMinifiedName: 'mat-toolbar-single-row', securityContext: undefined, suffix: undefined }) ], bindingFlags: 2, outputs: [  ], element: Object({ ns: '', name: 'mat-toolbar', attrs: [ Array, Array ], template: null, componentProvider: Object({ nodeIndex: 1, parent: <circular re ...
	    at <Jasmine>
	    at Object.../node_modules/jwt-decode/lib/index.js (http://localhost:9876/_karma_webpack_/vendor.js:112:31)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/app.guard.ts (http://localhost:9876/_karma_webpack_/main.js:3997:15257)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/navbar/navbar.component.ts (http://localhost:9876/_karma_webpack_/main.js:6266:28135)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at Object../src/app/app.component.spec.ts (http://localhost:9876/_karma_webpack_/main.js:3904:28)
	    at __webpack_require__ (http://localhost:9876/_karma_webpack_/main.js:80:30)
	    at webpackContext (http://localhost:9876/_karma_webpack_/main.js:960:9)
	    at <Jasmine>
	Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
	    at <Jasmine>
```

When it happens, it typically does on 1 of the 6 jobs (Linux + MacOS) that execute those tests:
![image](https://user-images.githubusercontent.com/3531020/69531919-0fddb600-0f75-11ea-9b16-80ace5ee19ee.png)

Re-running the broken job will at some point result in a passing job.

**Comment by bkimminich** [comment]
Should be resolved by 2c3dca4.

**Comment by bkimminich** [comment]
```
e 78.0.3904 (Mac OS X 10.13.3) ERROR
  An error was thrown in afterAll
  TypeError: Cannot read property 'substring' of undefined
      at <Jasmine>
      at SafeSubscriber._next (http://localhost:9876/_karma_webpack_/main.js:6266:31665)
      at SafeSubscriber.__tryOrUnsub (http://localhost:9876/_karma_webpack_/vendor.js:199416:16)
      at SafeSubscriber.next (http://localhost:9876/_karma_webpack_/vendor.js:199355:22)
      at Subscriber._next (http://localhost:9876/_karma_webpack_/vendor.js:199305:26)
      at Subscriber.next (http://localhost:9876/_karma_webpack_/vendor.js:199282:18)
      at Observable._subscribe (http://localhost:9876/_karma_webpack_/vendor.js:208670:20)
      at Observable._trySubscribe (http://localhost:9876/_karma_webpack_/vendor.js:198738:25)
      at Observable.subscribe (http://localhost:9876/_karma_webpack_/vendor.js:198724:22)
      at NavbarComponent.ngOnInit (http://localhost:9876/_karma_webpack_/main.js:6266:30724)
      at UserContext.<anonymous> (http://localhost:9876/_karma_webpack_/main.js:6163:19)
HeadlessChrome 78.0.3904 (Mac OS X 10.13.3): Executed 370 of 508 (skipped 1) ERROR (0 secs / 1 min 6.564 secs)
HeadlessChrome 78.0.3904 (Mac OS X 10.13.3) ERROR
  An error was thrown in afterAll
  TypeError: Cannot read property 'substring' of undefined
      at <Jasmine>
      at SafeSubscriber._next (http://localhost:9876/_karma_webpack_/main.js:6266:31665)
      at SafeSubscriber.__tryOrUnsub (http://localhost:9876/_karma_webpack_/vendor.js:199416:16)
      at SafeSubscriber.next (http://localhost:9876/_karma_webpack_/vendor.js:199355:22)
      at Subscriber._next (http://localhost:9876/_karma_webpack_/vendor.js:199305:26)
      at Subscriber.next (http://localhost:9876/_karma_webpack_/vendor.js:199282:18)
      at Observable._subscribe (http://localhost:9876/_karma_webpack_/vendor.js:208670:20)
      at Observable._trySubscribe (http://localhost:9876/_karma_webpack_/vendor.js:198738:25)
      at Observable.subscribe (http://localhost:9876/_karma_webpack_/vendor.js:198724:22)
      at NavbarComponent.ngOnInit (http://localhost:9876/_karma_webpack_/main.js:6266:30724)
      at UserContext.<anonymous> (http://localhost:9876/_karma_webpack_/main.js:6163:19)
e 78.0.3904 (Mac OS X 10.13.3): Executed 370 of 508 (skipped 1) ERROR (2 mins 52.746 secs / 1 min 6.564 secs)
npm ERR! Test failed.  See above for more details.
The command "if [[ -z "$TRAVIS_TAG" && "$TRAVIS_OS_NAME" != "windows" ]]; then NODE_ENV=test npm test; fi" exited with 1.
```

This error occured (instead?) during https://travis-ci.org/bkimminich/juice-shop/jobs/617943318 ... 😞
